src/main/scala/beam/router/skim/Skims.scala

```diff
@@ -2,43 +2,46 @@ package beam.router.skim
 
 import beam.router
 import beam.router.skim
-import beam.sim.BeamServices
+import com.google.inject.Inject
 import com.typesafe.scalalogging.LazyLogging
+import org.matsim.core.controler.MatsimServices
 
 import scala.collection.mutable
 
-object Skims extends LazyLogging {
+class Skims @Inject()(
+  matsimServices: MatsimServices,
+  odSkimmer: ODSkimmer,
+  tazSkimmer: TAZSkimmer,
+  driveTimeSkimmer: DriveTimeSkimmer
+) extends LazyLogging {
+
+  import Skims.SkimType
   lazy val od_skimmer: ODSkims = lookup(SkimType.OD_SKIMMER).asInstanceOf[ODSkims]
   lazy val taz_skimmer: TAZSkims = lookup(SkimType.TAZ_SKIMMER).asInstanceOf[TAZSkims]
   lazy val dt_skimmer: DriveTimeSkims = lookup(SkimType.DT_SKIMMER).asInstanceOf[DriveTimeSkims]
 
-  def setup(implicit beamServices: BeamServices): Unit = {
-    val skimConfig = beamServices.beamConfig.beam.router.skim
-    skims.put(SkimType.OD_SKIMMER, addEvent(new ODSkimmer(beamServices, skimConfig)))
-    skims.put(SkimType.TAZ_SKIMMER, addEvent(new TAZSkimmer(beamServices, skimConfig)))
-    skims.put(SkimType.DT_SKIMMER, addEvent(new DriveTimeSkimmer(beamServices, skimConfig)))
+  private val skims = mutable.Map.empty[SkimType.Value, AbstractSkimmer]
+  skims.put(SkimType.OD_SKIMMER, addEvent(odSkimmer))
+  skims.put(SkimType.TAZ_SKIMMER, addEvent(tazSkimmer))
+  skims.put(SkimType.DT_SKIMMER, addEvent(driveTimeSkimmer))
 
+  private def addEvent(skimmer: AbstractSkimmer): AbstractSkimmer = {
+    matsimServices.addControlerListener(skimmer)
+    matsimServices.getEvents.addHandler(skimmer)
+    skimmer
   }
 
-  def clear(): Unit = {
-    skims.clear()
```

Comment:
Was there a need for this clear? It seems like it would have been a cleanup needed elsewhere?

Reply:
Looking at its usage it seems only because it was a static object and this was a hack?

Reply:
Yes, they clear this because it's a static object and they need to explicitly clear it to free the memory.